### PR TITLE
Use sdk and clean up code

### PR DESCRIPTION
**Pull request by willemneal**
also add dockerfile and CI

<a href="https://gitpod.io/#https://github.com/nearprotocol/near-chess/pull/4"></a>

### CODE REVIEW

**Comment by willemneal** [comment]
```
 FAIL  src/test.js
  ● Test suite failed to run

    Can not sign transactions for account test.near, no matching key pair found in Signer.
      at Account.signAndSendTransaction (node_modules/nearlib/lib/account.js:63:19)
      at async Account.createAndDeployContract (node_modules/nearlib/lib/account.js:104:9)
      at async LocalTestEnvironment.setup (node_modules/near-shell/test_environment.js:33:9)
```

Any ideas @mikedotexe @amgando  @vgrichina @chadoh @janedegtiareva

**Comment by willemneal** [comment]
I thought I'd fix this here before moving it to near-examples since I had some local code I had forgotten to push and wanted a clean PR history in the new repo 🤓 .

File: .gitpod.yml [hunk 1]
```diff
@@ -1,5 +1,29 @@
+image:
+  file: .gitpod.Dockerfile
+# Options to prebuild the image after github events and set notifications/badges
+github:
+  prebuilds:
+    # enable for the master/default branch (defaults to true)
```
Comment (vgrichina) [inline .gitpod.yml:7]: I think we should just keep options that are actually needed and link to docs with full list of options, comments, etc.
It also looks like we are going a bit over the top here. Why have both `addComment` and `addBadge` enabled  for example? Or why pre-build non-PR branches?

File: .gitpod.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM gitpod/workspace-full
+
+RUN bash -c ". .nvm/nvm.sh \
+             && nvm install v12 && nvm alias default v12 \
+             && nvm use default && npm i -g yarn && alias near='yarn near'" \
```
Comment (vgrichina) [inline .gitpod.Dockerfile:5]: ` alias near='yarn near'` why is this needed?

File: assembly/main.ts [hunk 3]
```diff
@@ -37,14 +37,14 @@ export function createOrJoinGame(): void {
   let game: Game | null = null;
   if (lastId > 0) {
     gameKey = getGameKey(lastId);
-    game = Game.decode(storage.getBytes(gameKey));
-    if (game!.player2) {
+    game = storage.getSome<Game>(gameKey);
```
Comment (vgrichina) [inline assembly/main.ts]: should this use `getGame`?

File: package.json [hunk 4]
```diff
@@ -13,15 +13,15 @@
     "prestart": "npm run build:contract && npm run dev:deploy:contract",
     "start": "CONTRACT_NAME=$(cat neardev/dev-account) parcel src/index.html",
     "dev": "nodemon --watch assembly -e ts --exec 'npm run start'",
-    "test": "npm run build:contract && jest test --env=near-shell/test_environment"
+    "test": "npm run build:contract && npm run dev:deploy:contract && jest test"
```
Comment (vgrichina) [inline package.json]: `npm run dev:deploy:contract` why is this needed? Tests shouldn't be running on the same contract instance as `dev:deploy:contract` uses.

File: .gitpod.Dockerfile (hunk 2 above)
Comment (willemneal) [reply .gitpod.Dockerfile:5]: So that users can use `near` without installing it globally.

File: assembly/main.ts (hunk 3 above)
Comment (willemneal) [reply assembly/main.ts]: Yeah and use lastId.

File: .gitpod.Dockerfile (hunk 2 above)
Comment (vgrichina) [reply .gitpod.Dockerfile:5]: feel like it can be confusing. Why not just install `near` globally in image / pre-build anyway?

File: package.json (hunk 4 above)
Comment (mikedotexe) [reply package.json]: that was my mistake. meant to remove it. just pushed up a commit removing it.

File: .gitpod.Dockerfile (hunk 2 above)
Comment (mikedotexe) [reply .gitpod.Dockerfile:5]: I don't see what the problem is with this approach. Does this temporary Gitpod image really need a global dependency? The main thing is the command line acts the same as when they follow the more long-term instructions which tells them to install `near-shell`. Is this seen as blocking (a request to change) or a conversation about it?

Comment (vgrichina) [reply .gitpod.Dockerfile:5]: > The main thing is the command line acts the same as when they follow the more long-term instructions which tells them to install near-shell

So for this you want to have `yarn global add near-shell` in pre-build. Alias to locally installed behaves differently from globally installed. 

BTW, from my experience it doesn't work at all (raising question on how it was tested):
```
gitpod /workspace/near-chess $ near
bash: near: command not found
````

which makes sense, as `alias` is executed in context of building Docker image (and is active only in context of command passed with `bash -c`), not in context of GitPod's shell session.

Comment (mikedotexe) [reply .gitpod.Dockerfile:5]: This is what I did in an example I worked on that worked well:
https://github.com/near-examples/token-contract-as/blob/master/.gitpod.yml#L26
`alias near=./node_modules/near-shell/bin/near`
I believe I tried installing globally but ran into issues, but I'll try again tomorrow.

Comment (vgrichina) [reply .gitpod.Dockerfile:5]: so yeah, that example does it in completely different place (not when building Docker image).

Comment (willemneal) [reply .gitpod.Dockerfile:5]: The reason not to install globally is that we want to tie near-shell to the local version.

Comment (vgrichina) [reply .gitpod.Dockerfile:5]: ` && alias near='yarn near'` in Dockerfile does nothing though @willemneal

File: .gitpod.yml [hunk 5]
```diff
@@ -1,5 +1,24 @@
...
+    # add a check to pull requests (defaults to true)
+    addCheck: true
+    # add a "Review in Gitpod" button as a comment to pull requests (defaults to false)
+    addComment: true
+
+# List the start up tasks. You can start them in parallel in multiple terminals. See https://www.gitpod.io/docs/config-start-tasks/
 tasks:
-  - init: yarn
+  - before: nvm use default
+    init: yarn && alias near=./node_modules/near-shell/bin/near
```
Comment (vgrichina) [inline .gitpod.yml:21]: @willemneal I think might be more robust to use `$(yarn bin)/near` to avoid hardcoding path
